test/test.c

```diff
@@ -198,4 +208,45 @@ void raise_err(char *err_msg) {
    fputs(err_msg, stderr);
    putc('\n', stderr);
    exit(EXIT_FAILURE);
+}
+
+/* converts '\n' to replace_str */
+char *convert_space(char *src, char *replace_str) {
+   /* Scan the whole string, counting the occurrence of \n. */
+   int src_idx, count;
+
+   src_idx = 0;
+   count = 0;
+   while (src[src_idx] != '\0')
+      if (src[src_idx++] == '\n')
+         count++;
+
+   int replace_str_len;
+   int dest_len;
+   char *dest;
+
+   replace_str_len = strlen(replace_str);
+   dest_len = 1 + strlen(src) - count + count * replace_str_len;
+   dest = malloc(dest_len);
+   if (dest == NULL)
+      raise_err("test: failed to malloc.");
+   dest[dest_len - 1] = '\0';
+
+   int dest_idx;
+
+   src_idx = 0;
+   dest_idx = 0;
+   while (src[src_idx] != '\0')
+      if (src[src_idx] == '\n') {
+         strncat(dest + dest_idx, replace_str, replace_str_len);
```

Comment:
This line is disastrous! Since, by definition, `strncat` appends its second argument to **the end** of its first argument. Currently, however, `dest` has not been initialized to 0--it's just merely `malloc`ed; thus we do not know where would be the end, i.e. `'\0'`! Hence a catastrophic memory error arises. In order to fix this:

```c
dest = calloc();
strncat(dest);
// or
strncpy(dest + dest_idx);
```